start_bundle_coupled.R

```diff
@@ -260,61 +267,65 @@ for(scen in common){
+  already_rem <- mixedsort(Sys.glob(suche))[1]
+
   if (! is.na(already_rem)) {
-    if (! file.exists(already_rem)) stop(possibleRemindReport, " exists, but ", already_rem, " not!")
     iter_rem <- as.integer(sub(".*rem-(\\d.*)/.*","\\1", already_rem))
-  } else if (! is.na(scenarios_coupled[scen, "oldrun"])) {
+  } else {
     message("Nothing found for ", suche, ", continue with oldrun.")
-    if (isTRUE(stringr::str_sub(scenarios_coupled[scen, "oldrun"], -14, -1) == "/fulldata.gdx") &&
-        file.exists(scenarios_coupled[scen, "oldrun"])) {
-          already_rem <- c(scenarios_coupled[scen, "oldrun"])
+    if (.isFileAndAvailable(scenarios_coupled[scen, "oldrun"], "/fulldata.gdx")) {
+      already_rem <- scenarios_coupled[scen, "oldrun"]
     } else {
-      needle <- scenarios_coupled[scen, "oldrun"]
-      suche <- paste0(path_remind_oldruns, prefix_oldruns, needle, "-rem-*/fulldata.gdx")
+      lookfor <- if (is.na(scenarios_coupled[scen, "oldrun"])) scen else scenarios_coupled[scen, "oldrun"]
+      suche <- paste0(path_remind_oldruns, prefix_oldruns, lookfor, "-rem-*/fulldata.gdx")
       already_rem <- mixedsort(Sys.glob(suche))[1]
     }
   }
-  if (is.na(already_rem)) {
-    message("Nothing found for ", suche, ", starting with ", runname, "-rem-1.")
-  } else {
+  
+  if (.isFileAndAvailable(settings_remind[scen, "path_gdx"], "/fulldata.gdx")) {
+    # if there is a REMIND gdx given in scenario_cofig or scenario_config_coupled, use it instead of the one found automatically
+    message("Using REMIND gdx specified in ", basename(path_settings_coupled)," or ", basename(path_settings_remind),": ", settings_remind[scen, "path_gdx"])
+  } else if (! is.na(already_rem)) {
     # if there is an existing REMIND run, use its gdx for the run to be started
     settings_remind[scen, "path_gdx"] <- normalizePath(already_rem)
     message("Found REMIND gdx here: ", normalizePath(already_rem))
+  } else {
+    message("Nothing found for ", suche, ", starting with ", runname, "-rem-1.")
   }
+  
   # is there already a MAgPIE run with this name?
   iter_mag <- 0
-  needle <- scen
-  suche <- paste0(path_magpie, "output/", prefix_runname, needle,"-mag-*/report.mif")
+  suche <- paste0(path_magpie, "output/", prefix_runname, scen,"-mag-*/report.mif")
   already_mag <- mixedsort(Sys.glob(suche))[1]
+  
   if (! is.na(already_mag)) {
     iter_mag <- as.integer(sub(".*mag-(\\d.*)/.*","\\1",already_mag))
-  } else if (! is.na(scenarios_coupled[scen, "oldrun"])) {
-    message("Nothing found for ", suche, ", continue with oldrun")
-    if (isTRUE(stringr::str_sub(scenarios_coupled[scen, "oldrun"], -14, -1) == "/report.mif") &&
-        file.exists(scenarios_coupled[scen, "oldrun"])) {
-          already_mag <- c(scenarios_coupled[scen, "oldrun"])
+  } else {
+    message("Nothing found for ", suche, ", continue with path_report")
+    if (.isFileAndAvailable(scenarios_coupled[scen, "path_report"], "/report.mif")) {
```

Comment:
So you prioritize older magpie runs over `path_report`, but prioritize `fulldata.gdx` specified in `path_gdx` over older REMIND runs? We should document that somewhere.

Reply:
Okay, I changed this now: `path_gdx` *and* `path_report` overwrite whatever was found.